src/WPGraphQL.php

```diff
@@ -640,33 +622,45 @@ public static function get_allowed_post_types( $output = 'names', $args = [] ) {
 			$allowed_post_type_names = apply_filters( 'graphql_post_entities_allowed_post_types', $post_type_names, $post_type_objects );
 
 			// Filter the post type objects if the list of allowed types have changed.
-			if ( $post_type_names !== $allowed_post_type_names ) {
-				// Maybe they're just out of order.
-				sort( $post_type_names );
-				sort( $allowed_post_type_names );
-
-				if ( $post_type_names !== $allowed_post_type_names ) {
-					$post_type_objects = array_filter($post_type_objects, function ( $obj ) use ( $allowed_post_type_names ) {
-						return in_array( $obj->name, $allowed_post_type_names, true );
-					});
+			$post_type_objects = array_filter( $post_type_objects, static function ( $obj ) use ( $allowed_post_type_names ) {
```

Comment:
The upcoming conditions can be simplified by early returning if `graphql_single_name` is empty.